src/components/include/sl.h

```diff
@@ -382,13 +412,22 @@ sl_cs_exit_schedule_nospin_arg(struct sl_thd *to)
 		if (unlikely(!pt)) t = sl__globals()->idle_thd;
 		else               t = sl_mod_thd_get(pt);
 	}
-	thdcap = t->thdcap;
-	prio   = t->prio;
 
+	if (t->type == SL_THD_AEP_TCAP || t->type == SL_THD_COMP_TCAP) {
+		assert(t->budget && t->period);
+
+		if (t->last_replenish == 0 || t->last_replenish + t->period <= now) {
+			t->last_replenish = now;
+
+			/* TODO: need to change logic for SNDCAP with tcap_delegate, and error handling */
+			if (cos_tcap_transfer(sl_thd_rcvcap(t), sl__globals()->sched_tcap, t->budget, t->prio)) assert(0);
```

Comment:
Transfers are only if the AEP or COMP thread has it's own TCAP. For the ones that use the current scheduler's tcap, we don't do it. So that's why I've 4 different types of AEPs.